src/main/java/org/broadinstitute/hellbender/engine/ReadWalker.java

```diff
@@ -131,4 +136,21 @@ protected final void onShutdown() {
         // Overridden only to make final so that concrete tool implementations don't override
         super.onShutdown();
     }
+
+    /**
+     * Create a common SAMFileWriter for use with ReadWalkers.
+     *
+     * @param outputFile    - if this file has a .cram extension then a reference is required. Can not be null.
+     * @param referenceFile - the reference source to use. Can not be null if a output file has a .cram extension.
+     * @param header        - header to be used for the output writer
+     * @param preSorted     - if true then the records must already be sorted to match the header sort order
+     * @return SAMFileWriter
+     */
+    protected SAMFileGATKReadWriter createCommonSAMWriter(
+            final File outputFile,
+            final File referenceFile,
```

Comment:
Don't make the caller pass in the reference manually -- use the reference in `GATKTool.referenceArguments.getReferenceFile()`. Throw an exception if `GATKTool.hasReference()` is false and outputFile has a .cram extension.